srcjs/output_binding_html.js

```diff
@@ -107,23 +107,25 @@ function renderDependency(dep) {
   }
 
   if (dep.stylesheet) {
-    var stylesheets = $.map(asArray(dep.stylesheet), function(stylesheet) {
+    var link = $("<link rel='stylesheet' type='text/css'>");
+    $.map(asArray(dep.stylesheet), function(stylesheet) {
       var this_href = href + "/" + encodeURI(stylesheet);
+      $head.append(
+        // Force client to re-request the stylesheet if we've already seen it
+        // (without this unique param, the request might get cached)
+        link.attr("href", restyle ? (this_href + "?restyle=" + new Date().getTime()) : this_href)
+      );
+      if (!restyle) return;
       // If a styleSheet with this href is already active, then disable it
-      var re = new RegExp(this_href + "$");
+      var re = new RegExp(this_href.split("/").join("\\/") + "$");
       for (var i = 0; i < document.styleSheets.length; i++) {
         var h = document.styleSheets[i].href;
         if (!h) continue;
-        if (restyle && h.match(re)) document.styleSheets[i].disabled = true;
+        if (h.split("?restyle=")[0].match(re)) {
+          setTimeout(function() { document.styleSheets[i].disabled = true; }, 10);
```

Comment:
How about also adding `document.styleSheets[i].shiny_remove = true;`, and then after a `setTimeout(, 11)`, a function iterates over `document.styleSheets` and removes all items with that attribute?